### PR TITLE
Added contrbutors OG image using genratemetadata

### LINKED ISSUE #127: Add soical image for contributors profile page
It would have been nice if we had an og:image that show the image and name of the contributor;

image

example: 
image

### PR DESCRIPTION
**Pull request by dgparmar14**
# Description

Pull request for feature enhancements of contributor's OG image showcasing their merged PRs, reviewed PRs, and feed updates, akin to the highlight section. Additionally, it incorporates the contributor's GitHub image, GitHub username, and description as per their profile. The Open-Graph image also includes the organization logo.

fixes #127 

### Here are some examples of how it looks

Screenshot 2024-03-20 141803
Screenshot 2024-03-20 141743

### CODE REVIEW

**Comment by dgparmar14** [comment]
@rithviknishad Please review this pr and let's close the issue

File: public/logo.png
Comment (rithviknishad) [inline public/logo.png:1]: Organization specific data is not kept in this repo. Could you move this to the data repo like how we have the `logo.webp`? Also name this as `logo-black.png` instead. Or instead, you could store the color logo itself and instead use tailwind classes to make the image render as black using invert and brightness classes

File: app/api/og/route.tsx
Comment (rithviknishad) [inline app/api/og/route.tsx:1]: Use tailwindcss instead of css

File: app/api/og/Roboto-Bold.ttf
Comment (rithviknishad) [inline app/api/og/Roboto-Bold.ttf:1]: Use google fonts instead

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Accidentally clicked approve for previous review.... :/

**Comment by dgparmar14** [comment]
> https://vercel.com/docs/functions/og-image-generation/og-image-examples#using-tailwind-css---experimental

Sure I will do it,
I used that but I dropp the idea because of its still experimental,

**Comment by dgparmar14** [comment]
@rithviknishad, your requested changes have been implemented:

1. Changed `logo.webp` to `logo.jpg` and created a `symlink` for this.
2. Moved Roboto-Bold.ttf to the public directory since Google Font is not supported at the edge time.
3. Converted all possible CSS to Tailwind CSS in the OG route. However, please note that `Tailwind CSS` is experimental in og-images, so some styling remains in normal CSS.
4. Added metadataBase: new URL(env.NEXT_PUBLIC_META_URL) in the `layout.tsx` file to address the mentioned error.

Screenshot 2024-03-23 102106

Let me know if there's anything else you need!

File: app/api/og/route.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const pr_reviewed = parsedUrl.searchParams.get("pr_reviewed") || "0";
+  const eod_update = parsedUrl.searchParams.get("eod_update") || "0";
+
+  const logoImageData = await fetch(
+    new URL("../../../public/logo.jpg", import.meta.url),
+  ).then((res) => res.arrayBuffer());
+  const profileImage = `https://avatars.githubusercontent.com/${github_username}`;
+
+  const fontData = await fetch(
+    new URL("../../../public/Roboto-Bold.ttf", import.meta.url),
```
Comment (rithviknishad) [inline app/api/og/route.tsx:30]: contents in public directory can be accessed directly like this:
```suggestion
    new URL("/Roboto-Bold.ttf", import.meta.url),
```

image

**Review by rithviknishad — APPROVED** [review]

**Review by rithviknishad — CHANGES_REQUESTED** [review]
More such issues present... Try doing a local build and preview


image

**Comment by rithviknishad** [comment]
@dgparmar14 I assume whats happening is it is getting pre-rendered and so those files wouldn't be served until the build is completed and output is served right?

Have you tried importing the font from `next/fonts` instead? We could also disable pre-renders for this maybe.

**Comment by dgparmar14** [comment]
> @dgparmar14 I assume whats happening is it is getting pre-rendered and so those files wouldn't be served until the build is completed and output is served right?
> 
> Have you tried importing the font from `next/fonts` instead? We could also disable pre-renders for this maybe.

Yes I tried but it doesn't work at the edge time
And above error is because of first need to merge pr of data repo.

**Comment by dgparmar14** [comment]
> > @dgparmar14 I assume whats happening is it is getting pre-rendered and so those files wouldn't be served until the build is completed and output is served right?
> > 
> > Have you tried importing the font from `next/fonts` instead? We could also disable pre-renders for this maybe.
> 
> Yes I tried but it doesn't work at the edge time
> And above error is because of first need to merge pr of data repo.
> and i also run build locally and it is run perfectly fine with my changes.

**Comment by rithviknishad** [comment]
Reviewed the data-repo PR. Also it'd be great if you could mention related PR's from data-repo in this PR's body itself. It's nice to keep track of related PRs documented in the PR body.

**Comment by rithviknishad** [comment]
> and i also run build locally and it is run perfectly fine with my changes

Are you sure it's working locally? Not sure if I'm doing something wrong... I did checkout the data-repo to the one you've made and tried running this branch with it and getting the following errors upon visiting the profile of a contributor. Also getting 404 page not found for all contributors.

image

**Comment by dgparmar14** [comment]
> > and i also run build locally and it is run perfectly fine with my changes
> 
> Are you sure it's working locally? Not sure if I'm doing something wrong... I did checkout the data-repo to the one you've made and tried running this branch with it and getting the following errors upon visiting the profile of a contributor. Also getting 404 page not found for all contributors.
> 
> image
> 
> 

Sure, I will look out in the evening after exam

File: public/logo.png
Comment (rithviknishad) [inline public/logo.png:1]: Keep this file in the data repo instead. And update the loadOrgData script to copy it to this dir, similar to how the `logo.webp` is done.

**Review by rithviknishad — CHANGES_REQUESTED** [review]
LGTM, just a minor change and also merge conflicts needs to be resolved.

File: scripts/loadOrgData.js [hunk 2]
```diff
@@ -49,3 +49,16 @@ if (fs.existsSync(path.join(dataRepoPath, "config/assets"))) {
...
+  "app/contributors/[slug]",
+  "opengraph-image.tsx",
+);
+
+if (fs.existsSync(additionalFileSrc)) {
+  console.log("Copying additional file opengraph-image.tsx...");
+  fs.copyFileSync(additionalFileSrc, additionalFileDest);
+} else {
+  console.error("File opengraph-image.tsx not found in the config folder");
+}
```
Comment (rithviknishad) [inline scripts/loadOrgData.js:64]: Why copy the typescript file through load org data? It's not part of the data-repo right?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**